test/juttled/juttled.spec.js

```diff
@@ -440,6 +440,84 @@ describe('Juttled Tests', function() {
+                .then(function(response) {
+                    expect(response).to.have.status(200);
+
+                    var body = response.body;
+                    expect(body.is_root).to.be.true;
+                    expect(body.path).to.equal('/');
+                    expect(body.directories).to.deep.equal(['subdir']);
+                    expect(body.juttles.length).to.be.above(0)
+                })
+            });
+
+            it('list contents of subdir', function() {
+                return chakram.get(jd + '/directory?path=subdir')
+                .then(function(response) {
+                    expect(response).to.have.status(200);
+                    expect(response.body.is_root).to.be.false;
+                    expect(response.body.path).to.equal('/subdir');
+                })
+            });
+
+            it('list root using subdir + ..', function() {
+                return chakram.get(jd + '/directory?path=/subdir/..')
+                .then(function(response) {
+                    expect(response).to.have.status(200);
+                    expect(response.body.is_root).to.be.true;
+                    expect(response.body.path).to.equal('/');
+                })
+            });
+
+        });
+
+        describe('Invalid cases', function() {
+            it('non existant path 404s', function() {
+                var response = chakram.get(jd + '/directory?path=non-existent')
+                expect(response).to.have.status(404);
+                expect(response).to.have.json({
+                    code: 'JS-DIR-NOT-FOUND-ERROR',
+                    message: 'No such directory: non-existent',
+                    info: {
+                        path: 'non-existent'
+                    }
+                });
+                return chakram.wait();
+            });
+
+            it('trying to fetch child of root returns 403', function() {
+                var response = chakram.get(jd + '/directory?path=/../..');
+                expect(response).to.have.status(403);
+                expect(response).to.have.json({
+                    code: 'JS-DIR-ACCESS-ERROR',
+                    message: 'Can not read directory: ../..',
+                    info: {
+                        path: '../..'
+                    }
+                });
+                return chakram.wait();
+            });
+
+            it('fetching file and not directory throws error', function() {
+                var response = chakram.get(jd + '/directory?path=forever.juttle');
+                expect(response).to.have.status(404);
+                expect(response).to.have.json({
+                    code: 'JS-DIR-NOT-FOUND-ERROR',
+                    message: 'No such directory: forever.juttle',
+                    info: {
+                        path: 'forever.juttle'
+                    }
+                });
+                return chakram.wait();
+            });
+        });
+    });
+
```

Comment:
I'd also add a negative case where the path starts going down and then goes farther up, like /subdir/../../../. It should have the same result as immediately trying to go up, but I think it would be a useful second test depending on how you do the validation.